cmd/auth-handler.go

```diff
@@ -174,17 +225,36 @@ func checkRequestAuthType(ctx context.Context, r *http.Request, action policy.Ac
 		r.Body = ioutil.NopCloser(bytes.NewReader(payload))
 	}
 
-	if globalPolicySys.IsAllowed(policy.Args{
-		AccountName:     accountName,
-		Action:          action,
+	claims, s3Err := getClaimsFromToken(r)
+	if s3Err != ErrNone {
+		return s3Err
+	}
+
+	if accessKey == "" {
+		if globalPolicySys.IsAllowed(policy.Args{
+			AccountName:     accessKey,
+			Action:          action,
+			BucketName:      bucketName,
+			ConditionValues: getConditionValues(r, locationConstraint),
+			IsOwner:         false,
+			ObjectName:      objectName,
+		}) {
+			return ErrNone
+		}
+		return ErrAccessDenied
+	}
+
+	if globalIAMSys.IsAllowed(iampolicy.Args{
+		AccountName:     accessKey,
+		Action:          iampolicy.Action(action),
 		BucketName:      bucketName,
-		ConditionValues: getConditionValues(r, locationConstraint),
-		IsOwner:         isOwner,
+		ConditionValues: getConditionValues(r, ""),
 		ObjectName:      objectName,
+		IsOwner:         owner,
+		Claims:          claims,
 	}) {
 		return ErrNone
 	}
-
 	return ErrAccessDenied
```

Comment:
@harshavardhana, when MINIO_IAM_OPA_URL and MINIO_IAM_JWKS_URL are set, get an access denied in mc even to add a new config
```
➜  minio git:(604c4327f) ✗ export MINIO_IAM_OPA_URL=http://localhost:8181/v1/data/httpapi/authz
➜  minio git:(604c4327f) ✗ export MINIO_IAM_JWKS_URL=http://localhost:9763/oauth2/jwks         

➜  minio git:(604c4327f) ✗ ./minio server /tmp/fs     
➜  mc git:(master) ✗ mc config host add myminio http://192.168.1.167:9000 minio minio123
mc: <ERROR> Unable to initialize new config from the provided credentials. Access Denied.

Reply:
yes, @poornas that is the idea when OPA is set every request should be validated by OPA other than admin API.

Reply:
It is implemented as per requirement right now, I can change this to behave like if its Owner then to ignore OPA and continue @poornas

Reply:
ok, makes sense

Reply:
could you check with @abperiasamy @poornas - can change the requirement as needed..

Reply:
ok, will do

Reply:
@harshavardhana, checked with @abperiasamy - he confirmed that all request validations should be deferred to OPA when OPA is set.So, current approach is correct.